File: pythonProject/metodos.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def acelerar(self, quantidade):
        self.velocidade += quantidade

    def frear(self, quantidade):
        self.velocidade = max(0, self.velocidade - quantidade)

    def exibir_velocidade(self):
        print(f'Velocidade atual: {self.velocidade} km/h')

# Testando os métodos
carro = Carro("Ford", "Mustang", 2022)
carro.acelerar(50)
carro.exibir_velocidade()
carro.frear(20)
carro.exibir_velocidade()

File: GoLangProject/metodos.go
package main

import "fmt"

type Carro struct {
	Marca      string
	Modelo     string
	Ano        int
	Velocidade int
}

func (c *Carro) Acelerar(quantidade int) {
	c.Velocidade += quantidade
}

func (c *Carro) Frear(quantidade int) {
	c.Velocidade -= quantidade
	if c.Velocidade < 0 {
		c.Velocidade = 0
	}
}

func (c Carro) ExibirVelocidade() {
	fmt.Printf("Velocidade atual: %d km/h\n", c.Velocidade)
}

func main() {
	carro := Carro{Marca: "Ford", Modelo: "Mustang", Ano: 2022}
	carro.Acelerar(50)
	carro.ExibirVelocidade()
	carro.Frear(20)
	carro.ExibirVelocidade()
}
